lualib/resty/kong/peer_conn.lua

```diff
@@ -0,0 +1,50 @@
+local ffi              = require "ffi"
+local base             = require "resty.core.base"
+
+local orig_get_request = base.get_request
+local errmsg           = base.get_errmsg_ptr()
+local C                = ffi.C
+local ffi_str          = ffi.string
+local get_phase        = ngx.get_phase
+local NGX_ERROR        = ngx.ERROR
+
+local error            = error
+
+
+ffi.cdef[[
+int
+ngx_http_lua_kong_ffi_get_last_peer_connection_cached(ngx_http_request_t *r,
+    char **err);
+]]
+
+
+local function get_request()
+    local r = orig_get_request()
+
+    if not r then
+        error("no request found")
+    end
+
+    return r
+end
```

Comment:
This code is not reused anywhere, we can remove it and actually reduce the number of lines of code.